pkg/client/object.go

```diff
@@ -564,7 +564,12 @@ func (c *Client) getObjectHeaderV2(ctx context.Context, p *ObjectHeaderParams, o
 			)
 		}
 
-		hdr = v.GetHeader()
+		hdrWithSig := v.GetHeaderWithSignature()
+		if hdrWithSig == nil {
+			return nil, errors.New("got nil instead of header with signature")
+		}
+		hdr = hdrWithSig.GetHeader()
+		// todo: check signature there
```

Comment:
Are there any obstacles from doing it in this PR?

Reply:
The only obstacle is to release block for your PR as soon as possible. But it definitely can be fixed in this PR, I think. What do you think @cthulhu-rider ?

Reply:
With @cthulhu-rider we've decided to check signatures implicitly at type conversion (from unified structures like `v2` package to SDK structures). We will add it later as soon as SDK object structures will be finalized.